purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -672,26 +707,44 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         identityManager.currentAppUserID.let { appUserID ->
             purchases.forEach { purchase ->
                 if (purchase.containedPurchase.purchaseState == Purchase.PurchaseState.PURCHASED) {
-                    backend.postReceiptData(
-                        purchase.purchaseToken,
-                        appUserID,
-                        purchase.sku,
-                        allowSharingPlayStoreAccount,
-                        purchase.presentedOfferingIdentifier,
-                        !consumeAllTransactions,
-                        { info ->
-                            consumeAndSave(consumeAllTransactions, purchase)
-                            cachePurchaserInfo(info)
-                            sendUpdatedPurchaserInfoToDelegateIfChanged(info)
-                            onSuccess?.let { it(purchase, info) }
-                        }, { error, errorIsFinishable ->
-                            if (errorIsFinishable) {
+                    val postToBackend: (SkuDetails?) -> Unit = { skuDetails ->
```

Comment:
didn't make it a function to avoid having to pass all the stuff again and because I don't think it will be used outside of this method. I will extract it for readability.